src/main/java/Grapie/Functions/Storage.java

```diff
@@ -0,0 +1,199 @@
+            }
+
+            if (dataSplited.length == 3) {
+                //is a todo
+                Task task = new Todo(dataSplited[2]);
+                if (dataSplited[1].equals("O")) {
+                    task.isDone = true;
+                }
+                storingList.add(task);
+
+            } else if (dataSplited.length == 4) {
+                //is a event or deadline
+                if (dataSplited[0].equals("E")) {
+                    //event
+                    Event task = new Event(dataSplited[2], dataSplited[3]);
+                    if (dataSplited[1].equals("O")) {
+                        task.isDone = true;
+                    }
+                    storingList.add(task);
+
+                } else {
+                    //deadline
+                    Deadline task = new Deadline(dataSplited[2], dataSplited[3]);
+                    if (dataSplited[1].equals("O")) {
+                        task.isDone = true;
+                    }
+                    storingList.add(task);
+                }
+            }
+        }
+        return storingList; //return the filled list
+    }
+
+    /**
+     * Convert task into correct format, and store in dukeStorage.txt file.
+     *
+     * @param task The task to be converted into the new format.
+     * @param type Grapie.Tasks.Todo, Grapie.Tasks.Event or Grapie.Tasks.Deadline.
+     * @param time The date and time for the Tasks.
+     * @throws IOException Throws exception.
+     */
+    public void convertAndStore(Task task, String type, String time) throws IOException {
+        String doneOrNotDone = "";
+        if (task.isDone) {
+            doneOrNotDone += "O";
+        } else {
+            doneOrNotDone += "X";
+        }
+
+        String newDescription = "";
+        if (type.equals("T")) {
+            //todo
+            newDescription += type + " | " + doneOrNotDone + " | " + task.description;
+        } else {
+            //event & deadline
+            newDescription += type + " | " + doneOrNotDone + " | " + task.description + " | " + time;
+        }
+
+        File file = new File(filePath);
+        FileWriter fr = new FileWriter(file, true);
+
+        if (file.length() == 0) {
+            fr.write(newDescription);
+        } else {
+            fr.write("\n" + newDescription);
+        }
+
+        fr.close();
+    }
+
+    /**
+     * Modify the done value in the dukeStorage.txt file.
+     * Change not done (X) into done (O)
+     *
+     * @param lineNumber The line number to be deleted from hard disk.
+     * @throws IOException Throws exception.
+     */
+    public void replaceLineFromHardDisk(int lineNumber) throws IOException {
+        File myObj = new File(filePath);
+        Scanner myReader = new Scanner(myObj);
+
+        String newData = "";
+        int counter = 1;
+        while (myReader.hasNextLine()) {
+            String data = myReader.nextLine();
+            //System.out.println(data);
```

Comment:
don't retain commented out code; you can use VCS to retrieve it if you want at a later stage.